frontend/src/formdescriptioneditor/__tests__/FormDescriptionEditorWebSocketContainer.test.tsx

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { MockedProvider } from '@apollo/client/testing';
+import { render } from '@testing-library/react';
+import { FormDescriptionEditorWebSocketContainer } from 'formdescriptioneditor/FormDescriptionEditorWebSocketContainer';
+import React from 'react';
+import { Selection } from 'workbench/Workbench.types';
+
+jest.mock('uuid', () => ({ v4: () => '48be95fc-3422-45d3-b1f9-d590e847e9e1' }));
+
+const emptySelection: Selection = {
+  entries: [],
+};
+
+const emptySetSelection = (selection: Selection) => {};
+
+test('should render the FormDescriptionEditorWebSocketContainer', () => {
+  const { container } = render(
+    <MockedProvider>
+      <FormDescriptionEditorWebSocketContainer
+        editingContextId="editingContextId"
+        representationId="formDescriptionEditorId"
+        readOnly={false}
+        selection={emptySelection}
+        setSelection={emptySetSelection}
+      />
+    </MockedProvider>
+  );
+  expect(container).toMatchSnapshot();
```

Comment:
Please try to not use `toMatchSnapshot()` as much as possible and instead rely on proper tests. As an example, there are no test to ensure that the various drag and drop listeners are working as expected. It would probably provide a better user experience to support a simple click on the widgets to add them in the main area.

Reply:
Done